internal/ethereum/ethereum_service.go

```diff
@@ -195,6 +195,12 @@ func (service ethereumService) Update(dto *EthereumDto, node *ethereumv1alpha1.N
 	if dto.Storage != "" {
 		node.Spec.Storage = dto.Storage
 	}
+	if dto.Engine != nil {
+		node.Spec.Engine = *dto.Engine
+	}
+	if dto.JWTSecretName != "" {
+		node.Spec.JWTSecretName = dto.JWTSecretName
+	}
```

Comment:
move this inside, the `if dto.Engine` statement.
Because `jwtSecretName` won't be updated if `engine` is disabled.

check `if rpc` above for reference.